Draft-Accepted/RFCNNNN-Install-PowerShell.md

```diff
@@ -0,0 +1,177 @@
+
+## Shipping in Windows
+
+There are several issues that prevent shipping PowerShell 7 in Windows:
+
+- Size
+- Support lifecycle differences
+
+### Size constraints
+
+ A complete installation of PowerShell 7 currently is around 200 MB (unpacked). This is due to
+ including everything (modules and .NET assemblies) to provide parity with what Windows PowerShell
+ 5.1 with .NET Framework provides. There is currently no plan by Windows or the .NET team to ship
+ newer .NET in Windows, so PS7 will need to continue to ship .NET with PS7.
+
+### Support lifecycle
+
+ Windows has a 5+5 year support lifecycle. .NET supports up to 3 years for Long Term Support (LTS)
+ versions. So even if size was not an issue, we cannot check in a version of PS7 with a .NET that is
+ not supported while Windows itself is supported.
+
+### Solution
+
+To solve these issues, the proposal is to ship a new command in Windows and the PowerShell Gallery
+that would:
+
+- Be locally discoverable
+- Perform a simple download and install of the latest LTS release of PowerShell
+- Include parameter options to select Stable, Preview, or LTS releases
+
+## Goals/Non-Goals
+
+Goals:
+
+- By default, download and install the latest LTS release to the users CurrentUser Scope
+  - CurrentUser - default installation scope location $env:LOCALAPPDATA\Microsoft\pwsh
+  - AllUsers - Administrative users may explicit select this scope. AllUsers requires administrative
+    elevation and performs a silent install using the MSI defaults.
+- Allow users to use remoting for install at scale with Invoke-Command
+- Allow advanced users to specify channel Stable/Preview/LTS
+
+Non-goals:
+
+- Removal of PowerShell (Remove-PowerShell7)
+- Updating of PowerShell (Update-PowerShell7)
+- User selected versions prior to current (No version parameter)
+- Disconnected scenarios - Users that are air-gapped may download the appropriate release from GitHub.
+- Handling of additional MSI switch options - i.e. Enable PSRemoting
+- Cross platform support
+
+## Alternatives
+
+`install-powershell7` cmdlet is an alternative approach that is more common to discover for a
+Powershell user. The cmdlet approach increases the complexity of installing a new version of
+PowerShell from an existing version. Additionally, The ability to bootstrap and install PowerShell
+on a system that currently does not include Powershell is an added benefit for using a command.
+
+`install-powershell7` with -Scope CurrentUser installs PowerShell to $env:LOCALAPPDATA\Microsoft\pwsh. The user may wish to specify a location for the installation with a parameter **Path**.
+
+## Specification
+
+Command name: install-powershell7
+
+The default behavior is to install to CurrentUser scope, with the latest LTS
+release of PowerShell. If the user has administrative
+privilege, they can choose to install to the scope of AllUsers.
+
+- The package install location - $env:LOCALAPPDATA\Microsoft\pwsh
```

Comment:
If that directory already exists, shall the tool remove the existing folder first?

Reply:
added comment same here as above.  thank you